src/paulscode/android/mupen64plusae/input/map/PlayerMap.java

```diff
@@ -79,7 +88,9 @@ public String getDeviceSummary( Context context, int player )
             if( mMap.valueAt( i ) == player )
             {
                 int deviceId = mMap.keyAt( i );
-                String name = AbstractProvider.getHardwareName( deviceId );
+                String name = AbstractProvider.isHardwareAvailable( deviceId ) ?
+                        AbstractProvider.getHardwareName( deviceId ) :
+                        "Not Connected";
```

Comment:
This sounds like a bug to me. My intention was to deserialize the old ids if the device was still connected, and then serialize it using the device descriptor, without the old id in there. I will review the code and see if I can find anything that would cause this.